Emby.Server.Implementations/Syncplay/SyncplayController.cs

```diff
@@ -228,190 +271,271 @@ public void SessionJoin(SessionInfo session, JoinGroupRequest request)
             }
-            else if (request.Type.Equals(PlaybackRequestType.Seek))
+            else
             {
-                // Sanitize PositionTicks
-                var ticks = request.PositionTicks ??= 0;
-                ticks = ticks >= 0 ? ticks : 0;
-                if (_group.PlayingItem.RunTimeTicks != null)
-                {
-                    var runTimeTicks = _group.PlayingItem.RunTimeTicks ??= 0;
-                    ticks = ticks > runTimeTicks ? runTimeTicks : ticks;
-                }
+                // Client got lost, sending current state
+                var command = NewSyncplayCommand(SendCommandType.Pause);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
+            }
+        }
 
+        /// <summary>
+        /// Handles a seek action requested by a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The seek action.</param>
+        private void HandleSeekRequest(SessionInfo session, PlaybackRequest request)
+        {
+            // Sanitize PositionTicks
+            var ticks = SanitizePositionTicks(request.PositionTicks);
+
+            // Pause and seek
+            _group.IsPaused = true;
+            _group.PositionTicks = ticks;
+            _group.LastActivity = DateTime.UtcNow;
+
+            var command = NewSyncplayCommand(SendCommandType.Seek);
+            SendCommand(session, BroadcastType.AllGroup, command, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Handles a buffering action requested by a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The buffering action.</param>
+        private void HandleBufferingRequest(SessionInfo session, PlaybackRequest request)
+        {
+            if (!_group.IsPaused)
+            {
+                // Pause group and compute the media playback position
                 _group.IsPaused = true;
-                _group.PositionTicks = ticks;
-                _group.LastActivity = DateTime.UtcNow;
+                var currentTime = DateTime.UtcNow;
+                var elapsedTime = currentTime - _group.LastActivity;
+                _group.LastActivity = currentTime;
+                _group.PositionTicks += elapsedTime.Ticks > 0 ? elapsedTime.Ticks : 0;
+
+                _group.SetBuffering(session, true);
 
-                var command = NewSyncplayCommand(SendCommandType.Seek);
-                SendCommand(session, BroadcastType.AllGroup, command);
+                // Send pause command to all non-buffering sessions
+                var command = NewSyncplayCommand(SendCommandType.Pause);
+                SendCommand(session, BroadcastType.AllReady, command, CancellationToken.None);
+
+                var updateOthers = NewSyncplayGroupUpdate(GroupUpdateType.GroupWait, session.UserName);
```

Comment:
what if there are multiple sessions from same user, only one of which is buffering?

Reply:
This is merely a message for clients to know why the media is getting paused. Instead of showing a general "Someone is buffering" I thought of showing who's having problems.

A nice idea, for the future, would be to add a "group monitor tab" that shows the status of connected clients (user, device, buffering status, ping, etc.). In that case a more precise distinction would be needed.